MoreLinq.NoConflictGenerator/Program.cs

```diff
@@ -0,0 +1,227 @@
+                        case "-u":
+                        case "--using":
+                            usings.Add(Read(arg, MissingArgValue));
+                            break;
+                        case "":
+                            continue;
+                        default:
+                            dir = arg.Current[0] != '-'
+                                ? arg.Current
+                                : throw new Exception("Invalid argument: " + arg.Current);
+                            break;
+                    }
+                }
+            }
+
+            var includePredicate
+                = string.IsNullOrEmpty(includePattern)
+                ? (_ => true)
+                : new Func<string, bool>(new Regex(includePattern).IsMatch);
+
+            var excludePredicate
+                = string.IsNullOrEmpty(excludePattern)
+                ? (_ => false)
+                : new Func<string, bool>(new Regex(excludePattern).IsMatch);
+
+            var thisAssemblyName = typeof(Program).GetTypeInfo().Assembly.GetName();
+
+            var q =
+                from fp in Directory.EnumerateFiles(dir, "*.cs")
+                select new
+                {
+                    SourcePath = fp,
+                    Methods =
+                        from cd in
+                            CSharpSyntaxTree
+                                .ParseText(File.ReadAllText(fp), CSharpParseOptions.Default.WithPreprocessorSymbols("MORELINQ"))
+                                .GetRoot()
+                                .SyntaxTree
+                                .GetCompilationUnitRoot()
+                                .DescendantNodes().OfType<ClassDeclarationSyntax>()
+                        where (string) cd.Identifier.Value == "MoreEnumerable"
+                        from md in cd.DescendantNodes().OfType<MethodDeclarationSyntax>()
+                        let mn = (string) md.Identifier.Value
+                        where md.ParameterList.Parameters.Count > 0
+                        && md.ParameterList.Parameters.First().Modifiers.Any(m => (string)m.Value == "this")
+                        && md.Modifiers.Any(m => (string)m.Value == "public")
+                        && md.AttributeLists.SelectMany(al => al.Attributes).All(a => a.Name.ToString() != "Obsolete")
+                        && includePredicate(mn)
+                        && !excludePredicate(mn)
+                        select md
+                };
+
+            var indent = new string('\x20', 4);
+            var indent2 = indent + indent;
+            var indent3 = indent2 + indent;
+
+            var imports =
+                from nss in new[]
+                {
+                    new []
+                    {
+                        "System",
+                        "System.CodeDom.Compiler",
+                        "System.Collections.Generic",
+                    },
+                    usings.AsEnumerable(),
```

Comment:
Because the query comprehension syntax feels more natural to me (subjective), avoids lambda noise and so I try to stick to it as much as possible. Sometimes it might read longer, as in this case. Most of this program is expressed as query comprehensions so that's what I went with here too. If you find the following reads clearer then I can entertain the exception:

```c#
            var imports =
                new []
                {
                    "System",
                    "System.CodeDom.Compiler",
                    "System.Collections.Generic",
                }
                .Concat(usings.AsEnumerable())
                .Select(ns => indent + $"using {ns};");
```